### PR TITLE
Add search test

### PR DESCRIPTION
**Pull request by koppor**
This PR updates https://github.com/gabrielmldantas/jabref/pull/1 to the most resent JabRef code.

 * tests for search expressions
 * UI test for the search

This could help create tests for a solution for #1975

### CODE REVIEW

File: src/test/java/net/sf/jabref/logic/search/SearchQueryTest.java [hunk 1]
```diff
@@ -66,4 +66,30 @@ public void testSearchingForOpenBraketInBooktitle() {
...
+        entry.setField("author", "asdf");
+
+        assertFalse(new SearchQuery("qwer", true, true).isMatch(entry));
+        assertTrue(new SearchQuery("asdf", true, true).isMatch(entry));
+        assertTrue(new SearchQuery("author=asdf", true, true).isMatch(entry));
+    }
+
+    @Test
+    public void testIsValidQuery() {
+        assertFalse(new SearchQuery("asdf[", true, true).isValid());
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/search/SearchQueryTest.java]: Only one assert per test please..............

File: src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java [hunk 2]
```diff
@@ -58,4 +61,81 @@ public void testGetDatabaseFromMatchesDatabaseWithEntriesWithCorrectMatch() {
...
+        assertTrue(databaseSearcher.getMatches().isEmpty());
+    }
+
+    @Test
+    public void testGetDatabasefromMatches() {
+        BibDatabase database = new BibDatabase();
+        BibEntry entry = new BibEntry();
+        entry.setType(BibtexEntryTypes.INCOLLECTION);
+        entry.setField("author", "tonho");
+        database.insertEntry(entry);
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java]: Create test db in setup method and reuse it in the tests.

File: src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java [hunk 3]
```diff
@@ -58,4 +61,81 @@ public void testGetDatabaseFromMatchesDatabaseWithEntriesWithCorrectMatch() {
...
     }
+
+    @Test
+    public void testGetDatabasefromMatchesEmptyDatabase() {
+        BibDatabase database = new BibDatabase();
+        SearchQuery query = new SearchQuery("asdf", true, true);
+
+        DatabaseSearcher databaseSearcher = new DatabaseSearcher(query, database);
+
+        assertTrue(databaseSearcher.getMatches().isEmpty());
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java]: getMatches probably returns a list and thus it is better to compare against `Collections.emptyList()`

File: src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java [hunk 4]
```diff
@@ -58,4 +61,81 @@ public void testGetDatabaseFromMatchesDatabaseWithEntriesWithCorrectMatch() {
...
+        BibDatabase database = new BibDatabase();
+        BibEntry entry = new BibEntry();
+        entry.setType(BibtexEntryTypes.INCOLLECTION);
+        entry.setField("author", "tonho");
+        database.insertEntry(entry);
+
+        SearchQuery query = new SearchQuery("tonho", true, true);
+        DatabaseSearcher databaseSearcher = new DatabaseSearcher(query, database);
+
+        assertFalse(databaseSearcher.getMatches().isEmpty());
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java]: Similarily here, compare to `Collection.singletonList(expected hit)`

File: src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java [hunk 5]
```diff
@@ -58,4 +61,81 @@ public void testGetDatabaseFromMatchesDatabaseWithEntriesWithCorrectMatch() {
...
+    }
+
+    @Test
+    public void testGetDataBaseFromMatchesEmptyEntry() {
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/logic/search/DatabaseSearcherTest.java]: please rename the tests...the names are rather non-expressive. One should know from the name of a failing test what went wrong. see also naming convention in the wiki https://github.com/JabRef/jabref/wiki/Code-Howtos#test-cases

**Comment by tobiasdiez** [comment]
Just my usual remarks about tests: only one assert per test, expressive test names and `assertEquals(List, List)` instead of `assertTrue(list.isEmpty())`. 
Can be merged after fixing these comments.

**Comment by koppor** [comment]
@tschechlovdev thank you for the quick work

**Pull request merged**